src/emeus-vfl-parser.c

```diff
@@ -212,6 +213,39 @@ has_view (VflParser *parser,
   return g_hash_table_lookup (parser->views_set, name) != NULL;
 }
 
+/* Valid attributes */
+static const struct {
+  int len;
+  const char *name;
+} valid_attributes[] = {
+  { 5, "width" },
+  { 6, "height" },
+  { 7, "centerX" },
+  { 7, "centerY" },
+  { 3, "top" },
+  { 6, "bottom" },
+  { 4, "left" },
+  { 5, "right" },
+  { 5, "start" },
+  { 3, "end" },
+  { 8, "baseline" }
+};
+
+static char *
+get_offset_to (const char *str,
+               const char *tokens)
+{
+  char *offset = NULL;
+
+  for (int i = 0; i < G_N_ELEMENTS (tokens); i++)
```

Comment:
This is incorrect: `G_N_ELEMENTS` is a macro for stack allocated C arrays, not for heap allocated arrays or strings.

You'll have to use `strlen()` on the `tokens` argument to get its length.